Add ipk package builder script and makefile target

EmilioPeJu opened the pull request:
The IPK package will be used in the yocto rootfs.

EmilioPeJu committed:
Add ipk package builder script and makefile target

coretl reviewed (changes requested):
I guess that creating ipk files is manual. That seems to have an annoying amount of boilerplate, and will be used in at least 3 repos. I suggest:
- Putting `control` as a flat text file in the root of the repo (maybe named `ipk-control`?)
- Making a python package called `ipk-build` package on pypi that has a single `ipk-build` entrypoint that does the same as this script
- Installing it into the devcontainer along with all the other python deps
- Calling it from the makefile in this repo and other places that need to make ipk files

Please can you prototype this as a single python script `ipk-build.py` in this repo and we'll take it from there

EmilioPeJu commented:
Hi @coretl, the reason I did it this way is to avoid adding more dependencies, if you prefer to do so, we should use the project [opkg-util](https://git.yoctoproject.org/opkg-utils/) which was made for this task.

coretl commented:
I suggest adding it as a submodule and calling it from there then (it doesn't seem to be published on pypi)

EmilioPeJu committed:
Use opkg-utils to make the ipk package

The advantages are:
- The ipk build script is smaller.
- The metadata is validated.
The disadvantage is having an extra dependency which in this case is in
the form of a submodule.

EmilioPeJu commented:
OK, I've done a similar change to the one in the FPGA repo, the packaging related files are under `packaging` including the new submodule. The zpkg related part has been removed as requested, this doesn't allow to remove the PandABlocks-rootfs dependency because of some github-release script.

coretl commented:
> The zpkg related part has been removed as requested, this doesn't allow to remove the PandABlocks-rootfs dependency because of some github-release script.

I think that script is only used in this repo, because all the other GitHub releases are made by CI in GitHub. This is the only one that requires pushing locally, so unless you can find any other references in other repos you can move that script to this repo.

EmilioPeJu commented:
Done! it turned out that `make-githu-release.py` was removed from panda rootfs, I had to copy it from an old commit.
